src/bgw/job.h

```diff
@@ -16,6 +16,7 @@
 typedef struct BgwJob
 {
 	FormData_bgw_job fd;
+	bool unregister_check;
 } BgwJob;
```

Comment:
So this is now a kind of a weird thing -- a row from `bgw_job`, and at the same time arguments for `alter_job`. What would it mean if we pass empty `fd.check_name`, but at the same time `unregister_check = false`? I'd split these meanings.

Reply:
So I think it was not necessary to add this field after all, thank you for pointing it out.. I removed it. I also changed the behavior of alter_job: previously it would check the updated config with the old registered check function. Now it uses the new check function and the new config to perform the check.